### PR TITLE
Add test GitHub action

### PR DESCRIPTION
**Pull request by martinkennelly**
Signed-off-by: Kennelly, Martin <martin.kennelly@intel.com>

GA will fail until #331 is merged.

Issue: #327

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    pull_request:
+jobs:
+  build:
+    strategy:
+      matrix:
+        go-version: [1.13.x, 1.14.x, 1.15.x, 1.16.x]
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Set up Go 1.13
```
Comment (manuelbuil) [inline .github/workflows/test.yml:13]: The go version moves, right? So you should just add Set up go and remove the "1.13"

Comment (martinkennelly) [reply .github/workflows/test.yml]: Corrected.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+
+      - name: Install hwdata
+        run: sudo apt-get install hwdata -y
+
+      - name: Go test
```
[the remark this answers is not in the document]
Comment (martinkennelly) [reply .github/workflows/test.yml:24]: The reason why I did it was to ensure ```make test``` is ok for each version of golang. There is a small chance here to capture bugs for sure however it is free to execute this on github, and we just have to worry about log clutter only. I think that is an ok cost. Its easy to see at a glance if pass or failed.

**Comment by martinkennelly** [comment]
Combined into PR #328

**Pull request closed without merging**